opencog/learning/PatternMiner/PatternMiner.cc

```diff
@@ -1707,148 +1701,149 @@ void PatternMiner::calculateSurprisingness( HTreeNode* HNode, AtomSpace *_fromAt
-                // e.g.: M is the size of corpus
-                // A:  ( var_1 is from CAR ) && ( var_1 is horror ) , P(A) = 100/M
-                // A1: ( var_1 is from CAR ), A2: ( var_1 is horror )
-                // Ap: ( var_1 is from CAR ) && ( var_1 is horror ) && ( var_1 is male ) , P(Ap) = 99/M
-                // E:  ( var_1 is male )
-                // Different from the super pattern type two bellow, E adds one more condition to var_1, so P(Ap) should be < or = P(A).
-                // Surprisingness_II (A from Ap) =  min{|P(A) - P(Ap)*(P(Ai)/P(Ai&E))|} / P(A)
-                // Actually in this example, both A and Ap are quite interesting
-
-                // debug
-                cout << "For Super pattern: -------extended from a variable----------------- " << std::endl;
-                cout << unifiedPatternToKeyString(curSuperRelation.extendedHTreeNode->pattern, atomSpace);
-
-                float _min_surprisingness_II = 99999999.0f;
-
-                float p_Ap = (float)curSuperRelation.extendedHTreeNode->count / atomspaceSizeFloat;
-
-                // only calculate the connected subcomponets that also connected with patternE
-                for(unsigned int subgramNum = 1;  subgramNum < gram; ++ subgramNum)
-                {
+            // currently we only consider type 2
+//            if (curSuperRelation.isExtendedFromVar) // type one : extended from a variable,  the extended node itself is considered as a variable in the pattern A
+//            {
+//                // Ap is A's one supper pattern, E is the link pattern that extended
+//                // e.g.: M is the size of corpus
+//                // A:  ( var_1 is from CAR ) && ( var_1 is horror ) , P(A) = 100/M
+//                // A1: ( var_1 is from CAR ), A2: ( var_1 is horror )
+//                // Ap: ( var_1 is from CAR ) && ( var_1 is horror ) && ( var_1 is male ) , P(Ap) = 99/M
+//                // E:  ( var_1 is male )
+//                // Different from the super pattern type two bellow, E adds one more condition to var_1, so P(Ap) should be < or = P(A).
+//                // Surprisingness_II (A from Ap) =  min{|P(A) - P(Ap)*(P(Ai)/P(Ai&E))|} / P(A)
+//                // Actually in this example, both A and Ap are quite interesting
+
+//                // debug
+////                cout << "For Super pattern: -------extended from a variable----------------- " << std::endl;
+////                cout << unifiedPatternToKeyString(curSuperRelation.extendedHTreeNode->pattern, atomSpace);
+
+//                float _min_surprisingness_II = 99999999.0f;
+
+//                float p_Ap = (float)curSuperRelation.extendedHTreeNode->count / atomspaceSizeFloat;
+
+//                // only calculate the connected subcomponets that also connected with patternE
+//                for(unsigned int subgramNum = 1;  subgramNum < gram; ++ subgramNum)
+//                {
```

Comment:
you know -- a much easier way of commenting out code is just to put an #if 0 ... #endif around it.